src/battle_util.c

```diff
@@ -2773,13 +2773,24 @@ u8 DoBattlerEndTurnEffects(void)
                         gBattleCommunication[MULTISTRING_CHOOSER] = B_MSG_TERRAINPREVENTS_MISTY;
                         BattleScriptExecute(BattleScript_TerrainPreventsEnd2);
                     }
+                    else if (IsSleepClauseActiveForSide(GetBattlerSide(battler)))
+                    {
```

Comment:
open question to maintainers: we potentially need a new message to display for the following case in a double battle: 
T1: Player A uses Yawn on Opponent A (other moves this turn irrelevant)
T2: Player A uses Spore on Opponent B; end of turn, Opponent A should NOT fall asleep from Yawn because sleep clause was activated by Spore on Opponent A